staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -85,6 +85,12 @@ type store struct {
 	leaseManager        *leaseManager
 }
 
+func (s *store) RequestBookmark(ctx context.Context) error {
+	// Use watchContext to match ctx metadata provided when creating the watch.
+	// In best case scenario we would use the same context that watch was created, but there is no way access it from watchCache.
+	return s.client.RequestProgress(s.watchContext(ctx))
```

Comment:
IIUC we are just attempting to create similarly looking context, but no the same?

How safely can we rely on the fact that s.watchContext is enough?

Reply:
We are safe as this is based on grpc outgoing context metadata, which is not expected to be set outside of storage code.